### PR TITLE
Add support for Internal.AspNetCore.Sdk to be used as an msbuild-sdk package

### PR DESCRIPTION
**Pull request by natemcmaster**
We can define "SDKs" as packages. This gives us more flexibility and power than PackageReference. It also has the advantage of not putting a dummy package in the restore graph.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,7 @@ node_modules
 .nuget/NuGet.exe
 project.lock.json
 .build
+.idea/
```
Comment (ryanbrandenburg) [inline .gitignore]: Just curious, noticed this popping up in your PR's, what are you using IntelliJ for?

File: test/KoreBuild.FunctionalTests/SimpleRepoTests.cs [hunk 2]
```diff
@@ -74,16 +74,20 @@ public void FullBuildCompletes()
                     Assert.Collection(pkg.Descendants("File"),
                         a =>
                         {
-                            Assert.Equal("tools/any/any/Newtonsoft.Json.dll", a.Attribute("Path")?.Value);
+                            Assert.Equal("tools/any/netcoreapp2.1/Newtonsoft.Json.dll", a.Attribute("Path")?.Value);
```
Comment (ryanbrandenburg) [inline test/KoreBuild.FunctionalTests/SimpleRepoTests.cs]: Interesting that this should be a result, do we know why?

File: .gitignore (hunk 1 above)
Comment (natemcmaster) [reply .gitignore]: Rider :)

File: test/KoreBuild.FunctionalTests/SimpleRepoTests.cs (hunk 2 above)
Comment (natemcmaster) [reply test/KoreBuild.FunctionalTests/SimpleRepoTests.cs]: I switched the tests from using a .nuspec to using `PackAsTool`. When the tests were written, the SDK did not yet support packing tool projects.

**Pull request merged**